cachi2/core/models/output.py

```diff
@@ -1,80 +1,58 @@
-    """Metadata about a go-package dependency.
-
-    Unlike other dependency types, go-package dependencies may come from the standard library,
-    in which case their version is null.
+    Compliant to the CycloneDX specification:
+    https://cyclonedx.org/docs/1.4/json/#components
     """
 
-    type: Literal["go-package"]
+    name: str
     version: Optional[str]
 
+    type: Literal["library"] = "library"
 
-class PipDependency(_DependencyBase):
-    """Metadata about a pip dependency.
+    # This will soon be replaced by the purl
+    def key(self) -> str:
+        """Uniquely identifies a package.
 
-    Pip dependencies have a 'dev' value to indicate whether the dependency is used at runtime
-    or at build time (requirements.txt X requirements-build.txt).
-    """
-
-    type: Literal["pip"]
-    version: str
-    dev: bool
-
-
-Dependency = Annotated[
-    Union[GomodDependency, GoPackageDependency, PipDependency],
-    pydantic.Field(discriminator="type"),
-]
+        Used mainly for sorting and deduplication.
+        """
+        return f"{self.name}:{self.version}"
 
+    @classmethod
+    def from_package_dict(cls, package: dict[str, Any]) -> "Component":
+        """Create a Component from a Cachi2 package dictionary.
 
-def _dependency_sorting_key(dep: Dependency) -> tuple[str, bool, str, str]:
-    # should return all the attributes a Dependency (the whole Union) can have
-    return dep.type, getattr(dep, "dev", False), dep.name, dep.version or ""
+        A Cachi2 package has extra fields which are unnecessary and can cause validation errors.
+        """
+        return Component(name=package.get("name", None), version=package.get("version", None))
 
 
-class Package(pydantic.BaseModel):
-    """Metadata about a resolved package and its dependencies."""
+class Sbom(pydantic.BaseModel):
+    """Software bill of materials in the CycloneDX format.
 
-    type: PackageType
-    path: Path  # relative from source directory
-    name: str
-    version: str
-    dependencies: list[Dependency]
+    See full specification at:
+    https://cyclonedx.org/docs/1.4/json
+    """
 
-    @pydantic.validator("path")
-    def _path_is_relative(cls, path: Path) -> Path:
-        return check_sane_relpath(path)
+    bom_format: str = "CycloneDX"
+    components: list[Component] = []
+    spec_version: str = "1.4"
+    version: int = 1
 
-    @pydantic.validator("dependencies")
-    def _unique_deps(cls, dependencies: list[Dependency]) -> list[Dependency]:
-        """Sort and de-duplicate dependencies."""
-        return unique_sorted(dependencies, by=_dependency_sorting_key)
+    @pydantic.validator("components")
+    def _unique_components(cls, components: list[Component]) -> list[Component]:
+        """Sort and de-duplicate components."""
+        # Deduping might be a heavy hit on performance for larger requests
+        return unique_sorted(components, by=lambda component: component.key())
```

Comment:
Deduplication may have to be a lot more interesting, and possibly done already at the package manager level.

Consider https://github.com/redhat-appstudio/managed-gitops
```shell
$ grep -R '=> .*backend-shared'
appstudio-controller/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../backend-shared
backend/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../backend-shared
cluster-agent/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../backend-shared
tests-e2e/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../backend-shared
utilities/db-migration/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../../backend-shared
utilities/init-container/go.mod:	github.com/redhat-appstudio/managed-gitops/backend-shared => ../../backend-shared
```
"Different" version, same module.

Reply:
Although the 'deduplication at package manager level' may come with correct purl generation. We'll just have to get them right this time 😄